src/dll/fuse/fuse.c

```diff
@@ -871,6 +873,8 @@ FSP_FUSE_API struct fuse *fsp_fuse_new(struct fsp_fuse_env *env,
         opt_data.VolumeParams.VolumeInfoTimeoutValid = 1;
     if (opt_data.set_KeepFileCache)
         opt_data.VolumeParams.FlushAndPurgeOnCleanup = FALSE;
+    if (opt_data.set_FlushOnCleanup)
+        opt_data.VolumeParams.FlushOnCleanup = TRUE;
```

Comment:
Instead of placing the value of `FlushOnCleanup` flag in `VolumeParams` we should instead:

- Set `opt_data.VolumeParams.PostCleanupWhenModifiedOnly = !opt_data.set_FlushOnCleanup`. (Currently `opt_data.VolumeParams.PostCleanupWhenModifiedOnly` is initialized to `TRUE` further down in the code.)

- Add a `BOOLEAN FlushOnCleanup` to `struct fuse` and set `f->FlushOnCleanup = !!opt_data.set_FlushOnCleanup`.